crates/chain-listener/src/listener.rs

```diff
@@ -438,6 +549,35 @@ impl ChainListener {
         Ok(())
     }
 
+    async fn poll_proofs(&mut self) -> eyre::Result<()> {
+        if let Some(ref ccp_client) = self.ccp_client {
+            log::info!("Polling proofs after: {}", self.last_submitted_proof_id);
+
+            let proofs = ccp_client
+                .get_proofs_after(self.last_submitted_proof_id, PROOF_POLL_LIMIT)
+                .await?;
+
+            // TODO: send only in batches
+
+            // Filter proofs related to current epoch only
+            let proofs: Vec<_> = proofs
+                .into_iter()
+                .filter(|p| p.id.global_nonce == self.global_nonce)
+                .collect();
+
+            log::info!("Found {} proofs from polling", proofs.len());
+            for proof in proofs.into_iter() {
+                let id = proof.id.idx.clone();
+                log::info!("Submitting proof: {id}");
+                self.submit_proof(proof).await?;
+                log::info!("Proof submitted: {id}");
+                persist_proof_id(self.proof_id_dir.as_path(), id)?;
+                self.last_submitted_proof_id = id;
+            }
+        }
+        Ok(())
+    }
+
     async fn submit_proof(&mut self, proof: CCProof) -> eyre::Result<()> {
         match self.chain_connector.submit_proof(proof).await {
             Ok(_) => Ok(()),
```

Comment:
Let's log proof id.idx and tx_id to at least debug?

```suggestion
    async fn submit_proof(&mut self, proof: CCProof) -> eyre::Result<()> {
        let proof_id = proof.id.idx;
        match self.chain_connector.submit_proof(proof).await {
            Ok(tx_id) => {
                log::debug!("Submitted proof {proof_id}: {tx_id}");
                Ok(())
            },
```